newrelic-agent/src/main/java/com/newrelic/agent/service/logging/LogSenderServiceImpl.java

```diff
@@ -471,7 +472,9 @@ private static String mapInternString(String value) {
      * @return LogEvent instance
      */
     private static LogEvent createValidatedEvent(Map<String, ?> attributes) {
-        Map<String, Object> userAttributes = new HashMap<>(attributes.size());
+        // Initialize new userAttributes map with agent linking metadata required for LogEvents
+        Map<String, Object> userAttributes = new HashMap<>(AgentImpl.getLogEventLinkingMetadata());
```

Comment:
Is there any chance that there could be a thread switch from the log library instrumentation where we call `AgentBridge.getAgent().getLogSender().recordLogEvent(logEventMap)` to this point here where we actually process that API call and retrieve the linking metadata (`AgentImpl.getLogEventLinkingMetadata()`)? 

I don't think so, but if there was that would likely cause us to add the wrong linking metadata to the `LogEvent`. That would be my main concern with this refactoring.

Reply:
I'll admit I don't completely understand this question but I'd love to educated on this situation.  I would prefer the `logEventLinkingMetadata` be a parameter or pulled from an injected `Agent` rather than pulled from `AgentImpl` but I realize this is naive on my part and there is likely an issue with this.